DisguiseUnityRenderStream/Runtime/PipelineAbstraction/BlitExtended.cs

```diff
@@ -0,0 +1,73 @@
+using UnityEngine;
+using UnityEngine.Assertions;
+using UnityEngine.Rendering;
+
+class BlitExtended
+{
+    public enum ColorSpaceConversion
+    {
+        None,
+        LinearToSRGB,
+        SRGBToLinear
+    }
+    
+    static class ShaderIDs
+    {
+        public static readonly int _BlitTexture = Shader.PropertyToID("_BlitTexture");
+        public static readonly int _BlitScaleBias = Shader.PropertyToID("_BlitScaleBias");
+    }
+    
+    static BlitExtended s_Instance;
+
+    public static BlitExtended Instance
+    {
+        get
+        {
+            if (s_Instance == null)
+                s_Instance = new BlitExtended();
+
+            return s_Instance;
+        }
+    }
+
+    public static Vector4 IdentityScaleBias { get; } = new Vector4(1f, 1f, 0f, 0f);
+    public static Vector4 FlippedYScaleBias { get; } = FlipYScaleBias(IdentityScaleBias);
+    public static Vector4 FlipYScaleBias(Vector4 scaleBias) => new Vector4(scaleBias.x, -scaleBias.y, scaleBias.z, 1f - scaleBias.w);
+
+    static int GetPass(ColorSpaceConversion conversion)
+    {
+        return (int)conversion;
```

Comment:
I stole the pattern from `Blitter.cs` and `Blit.hlsl`.

The `int pass` corresponds to the index of the corresponding `Pass { ... }` block in the `BlitExtendedHDRP.shader` and `BlitExtendedURP.shader` file. Each pass has different vertex or fragment functions for the behavior it represents.

Passes can optionally be named and the index queried by name ([example](https://docs.unity3d.com/Manual/SL-Name.html)). I didn't do this to avoid managing lots of matching string constants in the code.

Reply:
Ha ok I see.  I guess this is a Unity notion I was missing.  Thanks.